src/Zicht/Itertools/lib/CollapseIterator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * @copyright Zicht Online <http://zicht.nl>
+ */
+
+namespace Zicht\Itertools\lib;
+
+use Zicht\Itertools\lib\Interfaces\FiniteIterableInterface;
+use Zicht\Itertools\lib\Traits\FiniteIterableTrait;
+
+class CollapseIterator extends \IteratorIterator implements FiniteIterableInterface
+{
+    use FiniteIterableTrait;
+
+    /**
+     * @param \Iterator $iterable
+     */
+    public function __construct(\Iterator $iterable)
+    {
+        $data = [];
+        foreach ($iterable as $subIterable) {
+            if (is_array($subIterable) || $subIterable instanceof \Traversable) {
+                foreach ($subIterable as $key => $value) {
+                    $data [] = [$key, $value];
```

Comment:
`$data[] = [$key, $value];`

Reply:
That syntax is being used throughout this entire library... and the linter mentions nothing about it...